core/iwasm/libraries/lib-socket/inc/wasi_socket_ext.h

```diff
@@ -59,6 +59,18 @@ typedef struct __wasi_addr_t {
 
 typedef enum { INET4 = 0, INET6 } __wasi_address_family_t;
 
+typedef struct __wasi_addr_info_t {
+    __wasi_addr_t addr;
+    __wasi_sock_type_t type;
+} __wasi_addr_info_t;
+
+typedef struct __wasi_addr_info_hints_t {
+    __wasi_sock_type_t type;
+    __wasi_address_family_t family;
+    // this is to workaround lack of optional parameters
+    uint8_t hints_enabled;
+} __wasi_addr_info_hints_t;
```

Comment:
Have we checked whether the PR works in 32-bit platform? Or the layout is same in both 32-bit and 64-bit platforms?

Reply:
I'm testing my changes on 64-bit Ubuntu 20.04 and 32-bit Lubuntu 18.04 (through qemu). It worked well on both.

Reply:
OK